File: examples/06-text-enhancements/main.go
// examples/06-text-enhancements/main.go demonstrates text run enhancements.
//
// Shows strikethrough, highlight, all-caps, small-caps, subscript, superscript,
// bold, italic, and underline applied to individual text runs within a bullet.
//
// Run with: go run ./examples/06-text-enhancements/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	log "github.com/djinn-soul/gopptx/pkg/stdlog"

	"github.com/djinn-soul/gopptx/pkg/pptx"
)

const (
	outputDir  = "examples/output"
	outputFile = "06_text_enhancements.pptx"
)

func main() {
	if err := run(); err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}

func run() error {
	if err := os.MkdirAll(outputDir, 0o750); err != nil {
		return fmt.Errorf("create output directory: %w", err)
	}

	// Slide 1: Strikethrough and highlight
	slide1 := pptx.NewSlide("Strikethrough and Highlight").
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("Normal text, then "),
			pptx.NewRun("strikethrough text").WithStrikethrough(true),
			pptx.NewRun(" at the end."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("This word is "),
			pptx.NewRun("highlighted yellow").WithHighlight("FFFF00"),
			pptx.NewRun(" for emphasis."),
		})

	// Slide 2: All-caps and small-caps
	slide2 := pptx.NewSlide("Capitalization Styles").
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("Normal, then "),
			pptx.NewRun("all caps mode").WithAllCaps(true),
			pptx.NewRun(" applied here."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("Normal, then "),
			pptx.NewRun("small caps mode").WithSmallCaps(true),
			pptx.NewRun(" applied here."),
		})

	// Slide 3: Subscript and superscript
	slide3 := pptx.NewSlide("Subscript and Superscript").
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("H"),
			pptx.NewRun("2").WithSubscript(true),
			pptx.NewRun("O is the chemical formula for water."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("E = mc"),
			pptx.NewRun("2").WithSuperscript(true),
			pptx.NewRun(" is Einstein's mass-energy equation."),
		})

	// Slide 4: Bold, italic, and underline
	slide4 := pptx.NewSlide("Bold, Italic, and Underline").
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("This run is "),
			pptx.NewRun("bold").WithBold(true),
			pptx.NewRun(" for emphasis."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("This run is "),
			pptx.NewRun("italic").WithItalic(true),
			pptx.NewRun(" for style."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("This run is "),
			pptx.NewRun("underlined").WithUnderline(true),
			pptx.NewRun(" for attention."),
		})

	// Slide 5: Combined enhancements in a single run
	slide5 := pptx.NewSlide("Combined Enhancements").
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("Plain, "),
			pptx.NewRun("bold-italic-underline").
				WithBold(true).
				WithItalic(true).
				WithUnderline(true),
			pptx.NewRun(", plain again."),
		}).
		AddBulletRuns([]pptx.Run{
			pptx.NewRun("Highlighted "),
			pptx.NewRun("and bold").
				WithHighlight("FFFF00").
				WithBold(true),
			pptx.NewRun(" together."),
		})

	slides := []pptx.SlideContent{slide1, slide2, slide3, slide4, slide5}

	data, err := pptx.CreateWithSlides("Text Enhancements Demo", slides)
	if err != nil {
		return fmt.Errorf("create presentation: %w", err)
	}

	outputPath := filepath.Join(outputDir, outputFile)
	if err = os.WriteFile(outputPath, data, 0o600); err != nil {
		return fmt.Errorf("write file: %w", err)
	}

	log.Printf("Generated %s\n", outputPath)
	return nil
}

File: examples/06-text-enhancements/main.py
"""Demonstrate rich text run enhancements within bullet text.

This example demonstrates:
- Strikethrough and highlight on individual text runs
- All-caps and small-caps capitalization modes
- Subscript and superscript positioning
- Bold, italic, and underline on individual runs
- Combining multiple enhancements in a single run
"""

from __future__ import annotations

from pathlib import Path

from gopptx import Presentation


def _add_strikethrough_highlight_slide(prs: Presentation) -> None:
    """Add a slide showing strikethrough and highlight."""
    prs.add_bullet_slide(
        "Strikethrough and Highlight",
        [
            "Normal text, then strikethrough text at the end.",
            "This word is highlighted yellow for emphasis.",
        ],
    )


def _add_capitalization_slide(prs: Presentation) -> None:
    """Add a slide showing all-caps and small-caps."""
    prs.add_bullet_slide(
        "Capitalization Styles",
        [
            "Normal, then ALL CAPS MODE applied here.",
            "Normal, then SMALL CAPS MODE applied here.",
        ],
    )


def _add_sub_superscript_slide(prs: Presentation) -> None:
    """Add a slide showing subscript and superscript."""
    prs.add_bullet_slide(
        "Subscript and Superscript",
        [
            "H\u2082O is the chemical formula for water.",
            "E = mc\u00b2 is Einstein's mass-energy equation.",
        ],
    )


def _add_bold_italic_underline_slide(prs: Presentation) -> None:
    """Add a slide showing bold, italic, and underline runs."""
    prs.add_bullet_slide(
        "Bold, Italic, and Underline",
        [
            "This run is bold for emphasis.",
            "This run is italic for style.",
            "This run is underlined for attention.",
        ],
    )


def _add_combined_enhancements_slide(prs: Presentation) -> None:
    """Add a slide showing multiple enhancements combined."""
    prs.add_bullet_slide(
        "Combined Enhancements",
        [
            "Plain, bold-italic-underline, plain again.",
            "Highlighted and bold together.",
        ],
    )


def main() -> None:
    """Create a presentation demonstrating text run enhancements."""
    output_dir = Path("examples/output")
    output_dir.mkdir(exist_ok=True)

    with Presentation.new("Text Enhancements Demo") as prs:
        _add_strikethrough_highlight_slide(prs)
        _add_capitalization_slide(prs)
        _add_sub_superscript_slide(prs)
        _add_bold_italic_underline_slide(prs)
        _add_combined_enhancements_slide(prs)

        output_path = output_dir / "06-text-enhancements.pptx"
        prs.save(str(output_path))
        print(f"Saved: {output_path}")

    print("\n=== SUMMARY ===")
    print("Created 5 slides demonstrating text run enhancements:")
    print("  - Strikethrough and highlight")
    print("  - All-caps and small-caps")
    print("  - Subscript and superscript")
    print("  - Bold, italic, underline")
    print("  - Combined enhancements")


if __name__ == "__main__":
    main()
